ml/Host.cc

```diff
@@ -437,6 +490,41 @@ void DetectableHost::detect() {
         updateDetectionChart(getRH(), Dur.count() * 1000);
 
         std::this_thread::sleep_for(Seconds{updateEvery()});
+
+        /*control the size of the database table and shrink them if required*/
+        //DB.shrinkAnomalyRateInfoTable(static_cast<int>(Cfg.MaxAnomalyRateInfoTableSize));
+        /*control the age of the data and remove them if required*/
+        //time_t OldestTime = now_realtime_sec() - (Cfg.MaxAnomalyRateInfoAge * 3600);
+        //DB.removeOldAnomalyRateInfo(OldestTime);
+    }
+}
+
+void DetectableHost::getAnomalyRateInfoCurrentRange(std::vector<std::pair<std::string, double>> &V, time_t After, time_t Before) {
+    for (auto &DP : DimensionsMap) {
+        Dimension *D = DP.second;
+        V.push_back({D->getID(), (D->getAnomalyPercentage() * abs(Before - After) / (Cfg.SaveAnomalyPercentageEvery * static_cast<double>(updateEvery())))});
+    }
+}
+
+void DetectableHost::getAnomalyRateInfoMixedRange(std::vector<std::pair<std::string, double>> &V, std::string HostUUID,time_t After, time_t Before) {
+    std::vector<std::pair<std::string, double>> DimAndAnomalyRateInRange;
+    bool Res = getAnomalyRateInfoInRange(DimAndAnomalyRateInRange, getUUID(), After, getLastSavedBefore());
+    std::vector<std::pair<std::string, double>>::iterator it;
+    
+    if (Res) {
+        for (auto &DP : DimensionsMap) {
+            Dimension *D = DP.second;
+            
+            /*Search in vector for corresponding dimension IDs, if found, combine and insert*/
+            auto it = std::find_if( DimAndAnomalyRateInRange.begin(), DimAndAnomalyRateInRange.end(),
+            [&D](const std::pair<std::string, int>& element){ return element.first == D->getID();} );
+            
+            if( it != DimAndAnomalyRateInRange.end())
+            {
+                double CurrentPercentage = (D->getAnomalyPercentage() * (Before - getLastSavedBefore()) / (Cfg.SaveAnomalyPercentageEvery * static_cast<double>(updateEvery())));
+                V.push_back({D->getID(), abs(((CurrentPercentage * (Before - getLastSavedBefore())) + (it->second * (getLastSavedBefore() - After)))/(Before - After))});
```

Comment:
Please assign the second element to a separate variable, it'll make reading the code somewhat easier.

Reply:
Done.

Reply:
I don't see this change in the latest PR changes.